### PR TITLE
Add support for the 'Connection Type' database to the geoip processor

### PR DESCRIPTION
**Pull request by joegallo**
Follow on to #107287, #107377, #108639, and #108651

Adds support for the ['GeoIP2 Connection Type'](https://dev.maxmind.com/geoip/docs/databases/connection-type) database from MaxMind to the `geoip` processor.

The `geoip` processor will automatically download the [various 'GeoLite2' databases](https://dev.maxmind.com/geoip/geolite2-free-geolocation-data), but the 'GeoIP2 Connection Type' database is not a 'GeoLite2' database -- it's a commercial database available to those with a suitable license from MaxMind.

The support that is being added for it in this PR is in line with the support that we already have for MaxMind's 'GeoIP2 City' and 'GeoIP2 Country' databases -- that is, one would need to arrange their own download management via some custom endpoint or otherwise arrange for the relevant file(s) to be in the `$ES_CONFIG/ingest-geoip` directory on the nodes of the cluster.

### CODE REVIEW

File: docs/reference/ingest/processors/geoip.asciidoc [hunk 1]
```diff
@@ -59,6 +59,8 @@ in `properties`.
 * If the GeoIP2 Anonymous IP database is used, then the following fields may be added under the `target_field`: `ip`,
 `hosting_provider`, `tor_exit_node`, `anonymous_vpn`, `anonymous`, `public_proxy`, and `residential_proxy`. The fields actually added
 depend on what has been found and which properties were configured in `properties`.
+* If the GeoIP2 Connection Type database is used, then the following fields may be added under the `target_field`: `ip`, and
+`connection_type`. The fields actually added depend on what has been found and which properties were configured in `properties`.
```
Comment (masseyke) [inline docs/reference/ingest/processors/geoip.asciidoc:63]: We need to also mention `connection_type` in the enterprise database section.

Comment (joegallo) [reply docs/reference/ingest/processors/geoip.asciidoc:63]: 1523e1fd6af0faab1ac14f0b2d90b609d9f5d925

**Review by masseyke — APPROVED** [review]
LGTM

**Comment by joegallo** [comment]
Related to https://github.com/elastic/elasticsearch/issues/101080

**Pull request merged**